packages/inventory/src/Stocktake/DetailView/columns.ts

```diff
@@ -2,42 +2,166 @@ import {
   GenericColumnKey,
   useColumns,
   ColumnAlign,
+  ifTheSameElseDefault,
+  formatExpiryDate,
+  Column,
   SortBy,
   SortRule,
 } from '@openmsupply-client/common';
-import { StocktakeItem } from './../../types';
+import { StocktakeLine, StocktakeSummaryItem } from './../../types';
 
 interface UseStocktakeColumnOptions {
-  sortBy: SortBy<StocktakeItem>;
+  sortBy: SortBy<StocktakeLine | StocktakeSummaryItem>;
   onChangeSortBy: (
-    newSortRule: SortRule<StocktakeItem>
-  ) => SortBy<StocktakeItem>;
+    newSortRule: SortRule<StocktakeLine | StocktakeSummaryItem>
+  ) => SortBy<StocktakeLine | StocktakeSummaryItem>;
 }
 
 export const useStocktakeColumns = ({
   sortBy,
   onChangeSortBy,
-}: UseStocktakeColumnOptions) =>
-  useColumns<StocktakeItem>(
+}: UseStocktakeColumnOptions): Column<StocktakeLine | StocktakeSummaryItem>[] =>
+  useColumns<StocktakeLine | StocktakeSummaryItem>(
     [
-      'itemCode',
-      'itemName',
-      'batch',
-      'expiryDate',
+      [
+        'itemCode',
+        {
+          getSortValue: row => {
```

Comment:
I don't think i'm following, but I certainly agree this area needs help, so am keen to find out!

- The shape: Yeah, I think having the basic shape is for the best and although adding the item name and code fields help for a couple of columns, a solution to make all of the columns better is surely possible.
- Helpers: Do you mean simply having `const getItemNameAccessor = (rowData) => ...` and using the same function for both the `getSortValue` and `accessor`?
- Constructor: Do you mean, for example, we could have in `useColumns`, the default `getSortValue` is equal to the `accessor`?

Reply:
I'm not following either.. sorry, not sure what was in my head when I wrote that.
Looking at it now - there's the duplication of code for the `accessor` and `getSortValue`, as we've talked about before. Makes sense to me that the sort value is the accessor value unless explicitly set otherwise.

Would be useful to have a helper as well, which returns for a given key either the value or the `ifTheSameElseDefault` value from another property ( 'lines' in this case ). Easy enough to write, a tricky one to type correctly I expect.